src/tail/tail.rs

```diff
@@ -101,6 +107,36 @@ pub fn uumain(args: Vec<String>) -> i32 {
         };
     }
 
+    match given_options.opt_str("pid") {
+        Some(pid_str) => {
+            let parsed: Option<i32> = pid_str.parse().ok();
+            match parsed {
+                Some(pid) => {
+                    settings.pid = pid;
+                    if pid != 0 {
+                        if !settings.follow {
+                            show_warning!("PID ignored; --pid=PID is useful only when following");
+                        }
+
+                        let unsupported = unsafe {
+                            if libc::kill(pid, 0) != 0 {
+                                errno::errno().0 == libc::ENOSYS
```

Comment:
I would recommend to fetch the last OS error code using `std::io::Error::last_os_error().raw_os_error().unwrap()` so that we don't need to introduce new crate here